core/modules/port_out.cc

```diff
@@ -109,6 +99,25 @@ void PortOut::ProcessBatch(Context *ctx, bess::PacketBatch *batch) {
     p->queue_stats[dir][qid].bytes += sent_bytes;
   }
 
+  return sent_pkts;
+}
+
+void PortOut::ProcessBatch(Context *ctx, bess::PacketBatch *batch) {
+  Port *p = port_;
+
+  queue_t qid = ctx->wid;
+  const queue_t num_queues = port_->num_queues[PACKET_DIR_OUT];
+
+  int sent_pkts = 0;
+  if (num_active_workers_ <= num_queues) {
+    sent_pkts = SendBatch(batch, p, qid);
+  } else {
+    mcslock_node_t me;
+    mcs_lock(&queue_locks_[qid], &me);
```

Comment:
I think this is a bug; each worker tries to hold a separate lock (`queue_locks_[qid]`, where each worker has a distinct `qid == ctx->wid` above)

Reply:
perhaps line 108 should be `queue_t qid = ctx->wid % num_queues`?

Reply:
oh yup! will fix